instrumentation/github.com/Shopify/sarama/otelsarama/producer_test.go

```diff
@@ -335,6 +332,99 @@ func TestWrapAsyncProducerError(t *testing.T) {
 	assert.Equal(t, "test", span.StatusMessage())
 }
 
+func TestWrapAsyncProducer_ConcurrencyEdgeCases(t *testing.T) {
+	newAsyncProducer := func() sarama.AsyncProducer {
+		cfg := newSaramaConfig()
+		var ap sarama.AsyncProducer = mocks.NewAsyncProducer(t, cfg)
+		ap = WrapAsyncProducer(cfg, ap) // you can comment this line to ensure that sarama is working in the same way
```

Comment:
Should we run these tests twice, once with a wrapped producer and once unwrapped?

Reply:
I think it makes sense.

Side note: Do you have any idea of other edge cases? I want to be more confident that the rewritten implementation does not introduce any new bugs.

Reply:
Done